presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkLocalExecutionPlanner.java

```diff
@@ -0,0 +1,219 @@
+                planFragment,
+                outputFactory,
+                createOutputPartitioning(taskContext, planFragment.getPartitioningScheme()),
+                remoteSourceFactory,
+                tableWriteInfo,
+                pageSinkCommitRequired);
+    }
+
+    @VisibleForTesting
+    public LocalExecutionPlan plan(
+            TaskContext taskContext,
+            PlanFragment planFragment,
+            OutputFactory outputOperatorFactory,
+            Optional<OutputPartitioning> outputPartitioning,
+            RemoteSourceFactory remoteSourceFactory,
+            TableWriteInfo tableWriteInfo,
+            boolean pageSinkCommitRequired)
+    {
+        Session session = taskContext.getSession();
+        PlanNode plan = planFragment.getRoot();
+        LocalExecutionPlanContext context = new LocalExecutionPlanContext(taskContext, tableWriteInfo);
+        PhysicalOperation physicalOperation = plan.accept(
+                new PrestoSparkVisitor(session, planFragment, remoteSourceFactory, pageSinkCommitRequired, taskSourceCodec, tableWriteInfoCodec, planFragmentCodec), context);
+
+        addDriverFactory(context, physicalOperation, outputOperatorFactory, planFragment.getPartitioningScheme(), outputPartitioning, plan, session);
+        addLookupOuterDrivers(context);
+
+        // notify operator factories that planning has completed
+        notifyPlanningCompletion(context);
+
+        return createLocalExecutionPlan(context, planFragment.getTableScanSchedulingOrder(), planFragment.getStageExecutionDescriptor());
+    }
+
+    private class PrestoSparkVisitor
+            extends Visitor
+    {
+        private final PlanFragment planFragment;
+        private final JsonCodec<TaskSource> taskSourceCodec;
+        private final JsonCodec<PlanFragment> planFragmentCodec;
+        private final JsonCodec<TableWriteInfo> tableWriteInfoCodec;
+
+        private PrestoSparkVisitor(
+                Session session,
+                PlanFragment planFragment,
+                RemoteSourceFactory remoteSourceFactory,
+                boolean pageSinkCommitRequired,
+                JsonCodec<TaskSource> taskSourceCodec,
+                JsonCodec<TableWriteInfo> tableWriteInfoCodec,
+                JsonCodec<PlanFragment> planFragmentCodec)
+        {
+            super(session, planFragment.getStageExecutionDescriptor(), remoteSourceFactory, pageSinkCommitRequired);
+            this.taskSourceCodec = requireNonNull(taskSourceCodec, "taskSourceCodec is null");
+            this.tableWriteInfoCodec = requireNonNull(tableWriteInfoCodec, "tableWriteInfoCodec is null");
+            this.planFragmentCodec = requireNonNull(planFragmentCodec, "planFragmentCodec is null");
+            this.planFragment = requireNonNull(planFragment, "planFragment is null");
+        }
+
+        @Override
+        public PhysicalOperation visitNativeExecution(NativeExecutionNode node, LocalExecutionPlanContext context)
```

Comment:
Would it possible to just have this implemented in `LocalExecutionPlanner`. There is no harm to just change it though it is only used for presto on spark. (Because we have already added NativeExecutionNode as a top-level citizen instead of being spark-only)